src/Model/Resolver/Products/Query/Search.php

```diff
@@ -153,6 +162,13 @@ public function getResult(
         $searchCriteria = $this->buildSearchCriteria($args, $info);
         $itemsResults = $this->getSearchResults($searchCriteria, $info);
 
+        if (!empty($args['filter']['category_id'])) {
+            $this->categoryCollectionFactory->create()
```

Comment:
Please add a detailed comment to explain what is happening here and the link to the task.